synapse/storage/data_stores/main/event_federation.py

```diff
@@ -72,13 +72,13 @@ def _get_auth_chain_ids_txn(self, txn, event_ids, include_given):
         while front:
             new_front = set()
             front_list = list(front)
-            chunks = [front_list[x : x + 100] for x in range(0, len(front), 100)]
+            chunks = (front_list[x : x + 100] for x in range(0, len(front), 100))
             for chunk in chunks:
                 clause, args = make_in_list_sql_clause(
                     txn.database_engine, "event_id", chunk
                 )
-                txn.execute(base_sql + clause, list(args))
-                new_front.update([r[0] for r in txn])
+                txn.execute(base_sql + clause, args)
```

Comment:
if we're relying on `make_in_list_sql_clause` returning a list, we should probably update its type annotation to say that's what it does?